scripts/rollup/shims/rollup/ReactCurrentOwner-www.js

```diff
@@ -0,0 +1 @@
+module.exports = require('ReactCurrentOwner');
```

Comment:
Note to self: I don't understand the purpose of  scripts/rollup/shims/rollup/*

Update: Looks like it maps to a combination of `forkedFBModules` and `getShims` in `modules.js`. Still not entirely sure why it's necessary. Seems like it's related some some quirk/limitation of Rollup?

Reply:
There should be comments close to getShims and its usage explaining.

The object-assign shim replaces object-assign reference in anything that depends on React with assign implementation exposed on React internals. This avoids ReactDOM UMD having to ship its own polyfill of object-assign while React UMD already includes one.

We use the www shims for overriding some modules in React. Basically we want to have some requires be "untouched" and actually require modules from www instead of our implementations in the open source. We can't just tell Rollup to turn `require('react/src/ReactCurrentOwner')` into `require('ReactCurrentOwner')` because then it will attempt to resolve it (and not find it). The trick with a shim file and a separate plugin to "transform" the original file to null in memory is the only reliable way I found to completely "rewire" these requires to external modules in Rollup.

Some of these hacks might fall away or become simpler after ES modules. CommonJS is afterthought in Rollup and it breaks some more unusual cases. It might be that we're in that territory, and migrating to ESM will allow to revisit and simplify those hacks.

Reply:
Yeah, I saw the comments. I just didn't fully understand them I guess. The third paragraph above helps me understand a little.